File: Swift/base-7.swift
class Solution {
    func convertToBase7(_ num: Int) -> String {
        if num < 0 {
            return "-" + convertToBase7(-num)
        }

        var res = ""
        var num = num

        while num >= 7 {
            res = String(num % 7) + res
            num /= 7
        }
        res = String(num) + res

        return res
    }
}

File: Python/base-7.py
class Solution:
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num < 0:
            return '-' + self.convertToBase7(-num)

        res = ''
        while num >= 7:
            res = str(num % 7) + res
            num //= 7
        res = str(num) + res

        return res
